### PR TITLE
fix instagram and youtube social embeds on AMP

### PR DESCRIPTION
**Pull request by DarioR01**
Resolves #N/A

**Overall change:**
Fix a bug where id's were not provided to Instagram and youtube embeds, causing the embeds to not display on AMP

**Code changes:**

- Created new youtube and Instagram embeds from the source

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [ ] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by amoore108 — APPROVED** [review]

**Review by RetiredBlob — APPROVED** [review]

File: src/app/legacy/containers/SocialEmbed/sourceHelpers.js [hunk 1]
```diff
@@ -35,9 +35,16 @@ export const getProviderFromSource = source => {
 export const getIdFromSource = source => {
   const NO_ID = '';
   const provider = getProviderFromSource(source);
-  if (provider === PROVIDERS.TWITTER) {
-    const match = source.match(/\/status\/([0-9]+)/);
-    return match ? match[1] : NO_ID;
+  if (
+    provider === PROVIDERS.TWITTER ||
```
Comment (eagerterrier) [inline src/app/legacy/containers/SocialEmbed/sourceHelpers.js]: not sure of our style guide, but could this be 

```
if ([PROVIDERS.TWITTER, PROVIDERS.YOUTUBE, PROVIDERS.INSTAGRAM].includes(provider)) {
```

File: src/app/legacy/containers/SocialEmbed/sourceHelpers.js [hunk 2]
```diff
@@ -35,9 +35,16 @@ export const getProviderFromSource = source => {
...
+    provider === PROVIDERS.YOUTUBE ||
+    provider === PROVIDERS.INSTAGRAM
+  ) {
+    const twitterId = source.match(/\/status\/([0-9]+)/);
+    const youtubeId = source.match(/\/watch\?v=([0-9 A-Z a-z _-]+)/);
+    const instagramId = source.match(/\/p\/([0-9 A-Z a-z]+)/);
```
Comment (eagerterrier) [inline src/app/legacy/containers/SocialEmbed/sourceHelpers.js]: hmm. Something can be done here, I just don't have the time or headspace to see what it is.

File: src/app/legacy/containers/SocialEmbed/sourceHelpers.js (hunk 1 above)
Comment (DarioR01) [reply src/app/legacy/containers/SocialEmbed/sourceHelpers.js]: Yes, I think no harm in doing this. I will update soon

File: src/app/legacy/containers/SocialEmbed/sourceHelpers.js (hunk 2 above)
Comment (DarioR01) [reply src/app/legacy/containers/SocialEmbed/sourceHelpers.js]: Here, I could have gone for an extended hard-to-read regex, I preferred splitting it for readability.
And I am sure there are some tricks that can be performed with arrays... But again, I don't think they will be as readable.

Comment (DarioR01) [reply src/app/legacy/containers/SocialEmbed/sourceHelpers.js]: I can use an object that holds the correct regex and call it with the provider e.g.
I think something like this won't harm readability:
```

const regex = {
  twitter: /\/status\/([0-9]+)/, 
  youtube: /\/watch\?v=([0-9 A-Z a-z _-]+)/, 
  Instagram: /\/p\/([0-9 A-Z a-z]+)/
}
const id = source.match(regex[provider]);
```

Comment (eagerterrier) [reply src/app/legacy/containers/SocialEmbed/sourceHelpers.js]: think I prefer this. You?

**Pull request merged**